code/modules/mob/living/carbon/carbon.dm

```diff
@@ -510,28 +501,57 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 		if(!A.pipe_image)
 			A.update_pipe_image()
 		pipes_shown += A.pipe_image
-		client.images += A.pipe_image
+		client?.images += A.pipe_image
 
 /mob/living/proc/remove_ventcrawl()
+	GLOB.ventcrawlers.Remove(src)
```

Comment:
Does `remove_ventcrawl` get called when the mob is destroyed? If not, there might be a `null` value in the ventcrawlers list.

Reply:
Basically yes. Update ventcrawl gets called every time a pipenet is updated, which will remove them. I added it to destroy to be sure.